src/coreclr/tools/aot/ILCompiler/Program.cs

```diff
@@ -238,6 +236,15 @@ public int Run()
                     }
                 }
 
+                foreach (var unmanagedEntryPointsAssembly in Get(_command.UnmanagedEntryPointsAssemblies))
+                {
+                    EcmaModule module = typeSystemContext.GetModuleForSimpleName(unmanagedEntryPointsAssembly);
+                    // Skip adding the system module as a UnmanagedEntryPointsRootProvider if it was already added as an input module
```

Comment:
I assume that this special-casing of system module is not required for functional correctness. It is just trying to avoid redundant work. Is that correct?

We can run into the same problem with other modules too. If you are concerned about redundant work, it would be better to have general test for duplicates here, do not special-case it for system module only.

Reply:
Correct. I modified the approach according to your suggestion.